engine.py

```diff
@@ -108,6 +116,57 @@ def post_app_init(self):
             except:
                 traceback.print_exc()
 
+    def validate_context(self, tank, context):
```

Comment:
I should have explained a bit more my thought. I also think it makes thing seems simpler at the first look, but since the engines custom startup logic is already in the `engine_init` hook for Nuke and Houdini, I thought it could be good to also have Katana in there. Anyway, you will have the last word on that, so it's up to you